lib/Service/Avatar/CompositeAvatarSource.php

```diff
@@ -34,12 +34,16 @@ class CompositeAvatarSource {
 	/** @var IAvatarSource[] */
 	private $sources;
 
-	public function __construct(AddressbookSource $addressbookSource,
-								FaviconSource $faviconSource,
-								GravatarSource $gravatarSource) {
+	public function __construct(
+		AddressbookSource $addressbookSource,
+		FaviconSource $faviconSource,
+		GravatarSource $gravatarSource,
+		BimiSource $bimiSource
```

Comment:
Keep the same order as you have below

```suggestion
		AddressbookSource $addressbookSource,
		BimiSource $bimiSource
		FaviconSource $faviconSource,
		GravatarSource $gravatarSource,
```